### ISSUE TITLE
Running two seperately compiled Lua projects.

### ISSUE THREAD

**Issue by NanoBob** [yanghuan/CSharp.lua#196]
**tldr;**
Would it be possible to support running multiple compiled projects together?

I have been using CSharp.lua in a project where we're working on creating Lua modules compiled by CSharp.lua. Our goal is to run these modules (as Lua) along with other Lua code compiled by CSharp.lua.

A little example:
We have a C# Project named `SlipeServer`, this project is compiled to a `SlipeServer` Lua "module". This is for all intents and purposes a library.
We have another C# Project name `ServerTest`, this project makes use of said library, and is also compiled to Lua. During this compilation the `.dll` file created by compiling the `SlipeServer` library is included in the compile process.
We wish to then run `ServerTest` and have that make use of `SlipeServer` in Lua.

However we have noticed several issues regarding this approach due to compiling only certain parts of the project. For example:
A `using` statement is normally compiled like this:

    local SlipeShared
    System.import(function (out)
      SlipeShared = Slipe.Shared
    end)

However when compiling our module and project seperately this instead results in

    local SlipeShared = Slipe.Shared

Which will cause an error since the Lua code defining `Slipe.Shared` has not yet executed. We've made a work around for this.

Another issue we encountered was the multiple manifest files, each separately compiled project resulted in another `manifest.lua` file as well. We've also made a workaround for this to allow loading multiple manifest files together.

The third issue we've encountered is one which would be much harder to resolve, and might even need some rather large changes. Namely constructor calls are not "translated" to their "proper" constructor IE: `__ctr1__` or `__ctr2__` for any constructors which are only in the DLL, and not in the C# source code

**tldr;**
Would it be possible to support running multiple compiled projects together?

**Comment by yanghuan** [comment]
thanks report, i will add a new cmd argument to control this,  it will tell the constructor calls from a dll may be stay same.

can you tell me, how you work around for System.import and manifest.lua ?

**Comment by NanoBob** [comment]
That's actually what we were thinking of as well, having a seperate option for dlls which are compiled to Lua.

For the System.import workaround we're treating all imports as if they are from code, we've also changed the `checkUsingDeclares` method. Both of these changes can be found in [this commit](https://github.com/mta-slipe/CSharp.lua/commit/44db073d02d44c5d3461f6e9798e4192320f0386) on our fork.


The workaround for the manifest however is a lot less elegant, the platform we're running our Lua code on does not support `require()`, instead all Lua files are loaded using `loadstring()`.
Our implementation can be seen in [this gist](https://gist.github.com/NanoBob/577e7e023da218183f1af8ef5a71e40f) however it is far from optimal.

**Comment by yanghuan** [comment]
“I suspect it might also impact regular overloaded methods”
yes, That's true.

“the platform we're running our Lua code on does not support require()”
Are you run Lua on browser ?

**Comment by NanoBob** [comment]
We're running Lua embedded in a game, http://mtasa.com

**Comment by yanghuan** [comment]
It is cool !

**Comment by yanghuan** [comment]
i commit 5b8445c34fb3b58db706b3acb447487fbe56a684 to support  multiple module compiled
Do you see if you meet your needs?

As for “using statement putout”
you can load the manifests.lua of  SlipeServer at first
then load  the manifests.lua of  ServerTest

**Comment by NanoBob** [comment]
Thanks for the quick response. The main functionality works however there are a couple issues:

It appears as if the information in the readme.md is incorrect. Instead of using the `-module` option as provided I was able to get it working by replacing `.dll` with `!` and still using the `-l` option. (CHecked source code to find out about this).

The second issue is regarding properties. On a class we have the property
```cs
public int R { get; set; }
```
This is compiled to a simple number on the class, not to the `getR()` and `setR()` methods. However when attempting to use a module compiled this way it does try to call the `setR()` and `getR()` methods. This would be able to be resolved to make properties with `{ get; set; }` compile the same way as properties with two methods for those.

Another issue that arose is regarding events.
We have the following event in a class in the library:
```cs
public delegate void OnVehicleDamageHandler(float loss);
public event OnVehicleDamageHandler OnDamage;
```

When adding an event handler from within that library the result is:
```lua
vehicle.OnDamage = vehicle.OnDamage + function (loss)
    System.Console.WriteLine("Loss")
end
```
However when adding an event handler from the project that uses the library, the result is:
```lua
vehicle:addOnDamage(function (loss)
	System.Console.WriteLine("Loss")
end)
```

**Comment by yanghuan** [comment]
dee6f76af4b66dd907463236a8ce721a5b257838
fix event & proeprty

**Comment by NanoBob** [comment]
When running this I get the following `FileNotFoundException`:
```
Could not load file or assembly 'System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

The stack trace traces this to the `IsModuleBackingFieldExists` method.

I've included the full stack trace and error in [this gist](https://gist.github.com/NanoBob/d70abbf8ae3716c27fd58bcb8b9ed287)

**Comment by yanghuan** [comment]
90566cd95899b5d19a196f33f78db20ef04c9c76 try to fix it

**Comment by NanoBob** [comment]
It works like a charm, Thanks so much for this update!
